bundle/direct/dresources/app_test.go

```diff
@@ -0,0 +1,76 @@
+package dresources
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/databricks/databricks-sdk-go/apierr"
+	"github.com/databricks/databricks-sdk-go/experimental/mocks"
+	"github.com/databricks/databricks-sdk-go/service/apps"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+)
+
+func TestDoCreate_RetriesOnAlreadyExists(t *testing.T) {
+	ctx := context.Background()
+	m := mocks.NewMockWorkspaceClient(t)
```

Comment:
I'd prefer to avoid mock client and use testserver instead. Even better would be to write acceptance test instead.